sw-precache.js

```diff
@@ -119,11 +120,20 @@ module.exports = function(params) {
   params.logger(util.format('Total precache size is about %s for %d resources.',
     formatBytesAsString(cumulativeSize), relativeUrls.length));
 
-  var templateBuffer = fs.readFileSync(params.templateFilePath);
-  return _.template(templateBuffer, {
+  var templateData = {
     handleFetch: params.handleFetch,
     importScripts: params.importScripts ? params.importScripts.map(JSON.stringify).join(',') : null,
     includeCachePolyfill: params.includeCachePolyfill,
     precacheConfig: JSON.stringify(precacheConfig)
-  });
+  };
+  var readStream = fs.createReadStream(params.templateFilePath, {encoding: 'utf8'});
+
+  return readStream.pipe(through(function(chunk, enc, done) {
+    if (chunk.isBuffer) {
+      this.emit('error', new Error('Buffers are not supported.'));
+    }
+
+    this.push(_.template(chunk, templateData));
+    done();
+  }));
```

Comment:
This is doing the exact thing I explained was wrong with `lodash-source-stream`: https://github.com/jeffposnick/sw-precache/pull/1#discussion_r22878263